packages/synapse-react-client/src/components/TanStackTable/ColumnHeaderEnumFilter.tsx

```diff
@@ -76,16 +92,22 @@ export function ColumnHeaderEnumFilter<TData = unknown, TValue = unknown>(
   // Use the column's enumValues and filterValues to create the facet value objects used by EnumFacetFilterUI
   const transformedFilterValues: RenderedFacetValue<TValue>[] = useMemo(
     () =>
-      enumValues.map(
-        (v: ColumnFilterEnumValue<TValue>): RenderedFacetValue<TValue> => {
+      getMaybeFacetedUniqueValues(column).map(
+        (facetValue): RenderedFacetValue<TValue> => {
+          let displayText: string = facetValue.value as unknown as string
+          if (column.columnDef?.meta?.getDisplayText) {
+            displayText = column.columnDef.meta.getDisplayText(facetValue.value)
+          }
```

Comment:
Get the display text (which was previously included in the `enumValues` data) using a new custom function.